README.md

```diff
@@ -28,6 +32,123 @@ Project is written in TypeScript and provides type safety out of the box. No Flo
 -   [Server Side Rendering with `useStaticRendering`](#server-side-rendering-with-usestaticrendering)
 -   [Why no Provider/inject?](#why-no-providerinject)
 
+## Overview
+
+This library has two goals:
+
+1. Make it possible to introduce reactivity to React function components
+2. Make it possible to have local observable state in React function components
+
+This library does _not_ support **class based components**. If you need class based component support, use [mobx-react](https://github.com/mobxjs/mobx-react), which wraps this library.
+
+### Adding reactivity to React components
+
+There are three ways to add reactivity to components: `observer(fn)`, `<Observer>{renderFn}</Observer>` and `useObserver(renderFn)`.
+In most cases, there difference doesn't matter that much, but here is an overview of the three different approaches:
+
+```javascript
+import { observable } from "mobx"
+import { Observer, useObserver, observer } from "mobx-react-lite"
+import ReactDOM from "react-dom"
+
+const person = observable({
+    name: "John"
+})
+
+// named function is optional (for debugging purposes)
+const P1 = observer(function P1({ person }) {
+    return <h1>{person.name}</h1>
+})
+
+const P2 = ({ person }) => <Observer>{() => <h1>{person.name}</h1>}</Observer>
+
+const P3 = ({ person }) => {
+    return useObserver(() => <h1>{person.name}</h1>)
+}
+
+ReactDOM.render(
+    <div>
+        <P1 person={person} />
+        <P2 person={person} />
+        <P3 person={person} />
+    </div>
+)
+
+setTimeout(() => {
+    person.name = "Jane"
+}, 1000)
+```
+
+[![Edit MobX in React - observer pattern](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/4qrv105l47?fontsize=14)
+
+When using `observer`, `React.memo` will be applied automatically to the component, and `useObserver` will be applied under the hood..
+
+`useObserver` keeps the component tree flat as it doesn't introduce a higher order component, but to make props observable, the [`useAsObservableSource`](#useasobservablesourcetstate-t-t) hook should be used.
+
+`<Observer>` is slightly better optimized compared to `useObserver`, as it doesn't cause a double rendering when using `useAsObservableSource`, but does introduce an additional component in the component tree.
```

Comment:
Not quite fan of `is slightly better optimized` as it's not quite exact from what I understand.

How about something along the lines of 

- `<Observer>` is basically just `useObserver` wrapped in a new component.
- `<Observer>` should be preffered in conjunction with `useAsObservableSource`, to avoid double rendering by creating component in the component tree.`
- avoid `useObserver` after `useAsObservableSource` as it will cause a `double rendering` issue, etc.